packages/utilities/src/asAsync.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+
+interface IAsAsyncOptions<TProps> {
+  /**
+   * Callback which returns a promise resolving an object which exports the component.
+   */
+  load: () => Promise<{ [key: string]: React.ReactType<TProps> }>;
+
+  /**
+   * The exportName to resolve from the result, defaults to "default".
+   */
+  exportName?: string;
```

Comment:
Is there a way to make this typesafe? Perhaps having load return the component instead of the entire module?

Reply:
```tsx
load: import('Dialog').then(result => result.default)
```

Feels a little weird, but maybe?

Reply:
the typesafety gets messier I think, if you have typdefs tslint rule enabled

Reply:
just a thought, I think it is fine like this if it gets messy